docs/design/settings/extensions/settings-extension-example.md

```diff
@@ -0,0 +1,112 @@
+* A setting extension crate called `mydaemon-settings` to add the new value `settings.mydaemon.query-url` to the API.
+* A `template` which will be rendered using the setting value and used as a configuration for the daemon.
+* A `service` file, which tells Bottlerocket how to handle the lifecycle of our daemon process as its settings change.
+
+## Creating the Settings Extension
+The easiest way to create an extension is to create a new Rust package in your project which utilizes the `bottlerocket-settings-sdk` crate.
+
+```
+├── packages
+├── sources
+│   └── mydaemon-settings
+│       ├── Cargo.toml
+│       └── src
+│           └── main.rs
+└── variants
+
+
+# sources/mydaemon/Cargo.toml
+[package]
+name = "mydaemon-settings"
+...
+[dependencies]
+bottlerocket-settings-sdk = 1.0
+...
+```
+
+Implementing the `bottlerocket_settings_sdk::Extension` trait provides a straightforward path to tell Bottlerocket how to validate and store the desired `query-url` setting.
+
+```rust
+#[derive(Serialize, Deserialize, Debug, Default, PartialEq, Clone)]
+struct MyDaemonSettingV1 {
+    query-url: Url,
+}
+
+impl SettingsModel for MyDaemonSettingV1 {
+    fn get_version() -> &'static str { /* */ }
+
+    fn set(current_value: Option<Self>, new_value: Self) -> Result<Self> { /* */ }
+    
+    // etc
+}
+```
+
+You can then refer to this model when invoking the `SettingsExtension` entrypoint to the SDK:
+
+```rust
+fn main() -> Result<()> {
+    bottlerocket_settings_sdk::SettingsExtension::with_models(vec![
+        BottlerocketSetting::<MyDaemonSettingsV1>::model(),
+    ])
+    .run_extension()
+    .context("Settings extension encountered an error.")
+}
+```
+
+
+## Configuration File Template
+In your variant repository, you will need to create an RPM package which installs `mydaemon` into your custom Bottlerocket variant.
+In this package, you will also want to include a templated configuration file, which is what the settings system interacts with:
+
+```
+├── packages
+│   └── mydaemon
+│       ├── build.rs
+│       ├── Cargo.toml
+│       ├── mydaemon.spec
+│       ├── mydaemon.template.yaml
+│       └── pkg.rs
+├── sources
+└── variants
+
+# mydaemon.template.toml
+---
+[required-extensions]
+mydaemon: v1
+---
+query-url: {{ mydaemon.query-url }}
+```
+The configuration template includes "frontmatter" which informs the settings system to use settings values owned by the `mydaemon` extension, then the rest of the template is used to render the configuration file.
+
+## Service File
+The settings system needs to know how to restart `mydaemon` so that it can parse changes to its config file after settings have been changed.
+In order to do that,you will create a `mydaemon-service.toml` file.
+This file will be referred to via a symlink on the Bottlerocket filesystem which instructs the settings system to restart the `mydaemon-service` when changes to `settings.mydaemon` take place:
```

Comment:
I find it hard to follow these files without a picture of the final install layout - the examples here tell me how to arrange the source files, but not what the destination layout should be, so I can't visualize the resulting spec file.

Reply:
I've given an example of how files will be arranged on the filesystem -- let me know if that seems ok to you.